File: daily/Linked List/725-Split-linked-list-in-parts.py
class Solution:
    def splitListToParts(self, head: Optional[ListNode], k: int) -> List[Optional[ListNode]]:
        ans = [None] * k
        size = 0
        current = head
        while current:
            size += 1
            current = current.next
        
        splitSize = size // k
        remainSize = size % k
        current = head
        for i in range(k):
            ans[i] = current
            currentSize = splitSize + (1 if remainSize > 0 else 0)
            remainSize -= 1

            for _ in range(currentSize - 1):
                if current:
                    current = current.next
            if current:
                nxt = current.next
                current.next = None
                current = nxt

        return ans

File: daily/Linked List/725-Split-linked-list-in-parts.go
package daily

func splitListToParts(head *ListNode, k int) []*ListNode {
	ans := make([]*ListNode, k)
	size := 0
	current := head
	for current != nil {
		size++
		current = current.Next
	}

	splitSize := size / k
	remainSize := size % k
	current = head
	for i := 0; i < k; i++ {
		ans[i] = current
		currentSize := splitSize
		if remainSize > 0 {
			currentSize++
			remainSize--
		}

		for j := 0; j < currentSize-1; j++ {
			if current != nil {
				current = current.Next
			}
		}
		if current != nil {
			nxt := current.Next
			current.Next = nil
			current = nxt
		}
	}

	return ans
}
